### DISCUSSION TITLE
Is it possible to use windows / ubuntu images in scale set on Azure

### DISCUSSION THREAD

**Discussion by facusantillo** [actions/virtual-environments#7296] [General questions] [answered]
As the title says, we want to move from hosted agents to scale sets but I'm not finding in the documentation if it's possible to use the images on the scale sets agents.

Anyone has run into it before?

**Comment by n-cow** [comment, marked as answer]
I have just done that for a client.  Images (win+linux) are working well at scale on multiple scale sets.

Clone the repo, build the image with Packer, push result to compute gallery, point a scaleset at that image, point ADO at that scale set.

Image takes us over 6 hours to build, so I set up weekly scheduled refreshes based on this repo.

I added some customisation steps to build on top of the base runners (add CAs, custom software etc.).  No major changes needed to the base image though.

**Reply by facusantillo** [reply]
Excellent, I'll take a look at it. Thanks!
